src/pages/AddLiquidity/index.tsx

```diff
@@ -89,15 +89,17 @@ export default function AddLiquidity({
 
   // keep track for UI display purposes of user selected base currency
   const [baseCurrency, setBaseCurrency] = useState(currencyA)
-  const quoteCurrency = useMemo(
-    () =>
-      currencyA && currencyB && baseCurrency
-        ? currencyEquals(baseCurrency, currencyA)
-          ? currencyB
-          : currencyA
-        : undefined,
-    [currencyA, currencyB, baseCurrency]
-  )
+  const quoteCurrency = useMemo(() => {
+    return currencyA &&
+      currencyB &&
+      baseCurrency &&
+      // need this for case where it evaluates on url change before baseCurrency is updated
+      (currencyEquals(baseCurrency, currencyA) || currencyEquals(baseCurrency, currencyB))
```

Comment:
why is baseCurrency being updated in a useEffect? isn't it just always currencyA?

Reply:
So to fix a different bug we need to reset it to undefined on dismount  - so for a url change Add component should dismount and set currencyA to undefined 

but for some reason in this useMemo the old baseCurrency value is being used on a url change (it then updates itselfa second time, so only a problem when weth / eth involved)

so i think its 

1. user selects new currency, url changes  
2. this useMemo tries to assign quoteCurrency with the old baseCurrency but new currencyB
3. new baseCurrency gets set from new currencyA
4. useMemo fires again and sets it